tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -866,29 +902,13 @@ private void OnHostingHttpRequestInStop(object arg)
 
             if (scope != null)
             {
-                var span = scope.Span;
-
                 // We may need to update the resource name if none of the routing/mvc events updated it.
                 // If we had an unhandled exception, the status code will already be updated correctly,
                 // but if the span was manually marked as an error, we still need to record the status code
-                var isMissingHttpStatusCode = !span.HasHttpStatusCode();
                 var httpRequest = arg.DuckCast<HttpRequestInStopStruct>();
                 HttpContext httpContext = httpRequest.HttpContext;
-                if (string.IsNullOrEmpty(span.ResourceName) || isMissingHttpStatusCode)
-                {
-                    if (string.IsNullOrEmpty(span.ResourceName))
-                    {
-                        span.ResourceName = AspNetCoreRequestHandler.GetDefaultResourceName(httpContext.Request);
-                    }
-
-                    if (isMissingHttpStatusCode)
-                    {
-                        span.SetHttpStatusCode(httpContext.Response.StatusCode, isServer: true, tracer.Settings);
-                    }
-                }
 
-                span.SetHeaderTags(new HeadersCollectionAdapter(httpContext.Response.Headers), tracer.Settings.HeaderTags, defaultTagPrefix: SpanContextPropagator.HttpResponseHeadersTagPrefix);
-                scope.Dispose();
+                DoBeforeRequestStops(httpContext, scope, tracer.Settings);
```

Comment:
AFAIK, this code is called even when there's an exception, so I don't think you need to call this function everywhere 🤔